plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -93,7 +93,7 @@ public class CachingJdbcClient
 
     @Inject
     public CachingJdbcClient(
-            @StatsCollecting JdbcClient delegate,
+            RetryingJdbcClient delegate,
```

Comment:
should it be injected with annotation?

Reply:
Actually it is messy here. I think it would be nice to clean it up. There different approaches used in this module :(

Reply:
This should fix the mess https://github.com/trinodb/trino/pull/23330